### PR TITLE
Separate out setting a cluster's TLS & hostname

### PR DESCRIPTION
**Pull request by yashgorana**
## Description

Closes https://github.com/OpenMined/Heartbeat/issues/813

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Comment by yashgorana** [comment]
tested this manually

## Step 1. basic http setup

Command `helm install syft ./helm/syft --namespace syft --set node.settings.nodeType=gateway --create-namespace`

Deploys ingress with name `grid-stack-ingress`
image

YAML
image

## Step 2. Upgrade to use TLS with domain
Command: `helm upgrade syft ./helm/syft --set node.settings.nodeType=gateway --set node.settings.tls=true --set node.settings.hostname=localhost.openmined.org --namespace syft`

Ingress swapped to `grid-stack-ingress-tls`
image

YAML
image

Should be pretty straight forward now =)

File: packages/grid/helm/helm.py [hunk 1]
```diff
@@ -140,6 +161,14 @@ def main() -> None:
 
     # Load the multi-doc yaml file
     try:
+        # append custom docs
+        input_data = "\n---\n".join(
+            [
+                input_data,
+                ingress_with_tls(),
```
Comment (rasswanth-s) [inline packages/grid/helm/helm.py:168]: NITPICK: We could add  it in devspace.yaml , in ingress deployment manifests?

Comment (yashgorana) [reply packages/grid/helm/helm.py:168]: Specifically not added it in devspace as it's a dev tool and primary use will be for local dev, so no point adding TLS config for dev. Plus we'll be moving away from generating helm charts from it. Additionally, devspace doesn't offer any such feature as well, unless we use profiles which generates a whole different chart.

Comment (rasswanth-s) [reply packages/grid/helm/helm.py:168]: makes sense

**Review by rasswanth-s — APPROVED** [review]
Stellar Work @yashgorana  ⭐

**Pull request merged**